packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -125,54 +118,37 @@ class PlotlyDivGraph extends CoreElement {
 
       if (!_processDatum) {
         // The once a second chunking (other chunking uses frameChunking).
-        timer = Timer.periodic(const Duration(seconds: 1), (Timer t) async {
-          // Get the lock.
-          await mutex.acquireWrite();
-          try {
-            plotData(timelineController);
-          } finally {
-            // Release the lock.
-            mutex.release();
-          }
+        timer =
+            Timer.periodic(const Duration(milliseconds: 166), (Timer t) async {
+          // Skip if there is no new data.
+          if (_lastPlottedFrameIndex == _frameIndex) return;
```

Comment:
The key is we don't plotData at all if no new data was added. We could be smart and pause the timer when no new data has been added but I'll leave that for a future cl.

Reply:
I like this much better.

Reply:
I don't see anything async within the body of the plotData or frameAdd code so we are safe.

Reply:
Jacob would you mind adding a todo for pausing the timer? Just so we can keep track.